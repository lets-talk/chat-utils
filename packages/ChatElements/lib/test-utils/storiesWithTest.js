import { withTests } from '@storybook/addon-jest';
import jestTestResults from '../../tests-results/jest-test-results.json';

export default withTests(jestTestResults, {
  filesExt: '__tests__/**.js',
});
