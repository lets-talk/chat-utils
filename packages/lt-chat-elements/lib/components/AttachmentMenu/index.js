import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { rgba } from 'polished';

import { Flex, Box } from 'grid-styled';

import MdImage from 'react-icons/lib/md/image';
import MdVideo from 'react-icons/lib/md/video-library';
import MdAudioTrack from 'react-icons/lib/md/audiotrack';
import MdLockOpen from 'react-icons/lib/md/lock-open';
import MdAttachFile from 'react-icons/lib/md/attach-file';

import { textColor, themeColor } from '../../utils/style';

const StyledMenuList = styled.ul`
  display: block;
  cursor: pointer;
  padding: 0;
  margin: 0;
  border-radius: 5px;
  background-color: ${(props) => props.theme.palette.common.white};
  font-size: ${(props) => props.theme.typography.classes.caption.fontSize};
  line-height: ${(props) => props.theme.typography.classes.caption.lineHeight};
  font-weight: ${(props) => props.theme.typography.weights.fontWeightMedium};
  box-shadow: ${(props) => props.theme.shadows['4p']};

  li {
    list-style: none;
    padding: 15px;
    color: ${(props) => textColor(props.theme, 'light', 'secondary')};

    &:hover {
      color: ${(props) => themeColor(props.theme, 'accent', 'base')};
      background-color: ${(props) => rgba(themeColor(props.theme, 'accent', 'base'), 0.1)};
    }
  }
`;

const renderIcon = (type, size) => {
  switch (type) {
    case 'image':
      return <MdImage size={size} />;
    case 'video':
      return <MdVideo size={size} />;
    case 'audio':
      return <MdAudioTrack size={size} />;
    case 'document':
      return <MdImage size={size} />;
    case 'link':
      return <MdLockOpen size={size} />;
    default:
      return <MdAttachFile size={size} />;
  }
};

const AttachmentMenu = (props) => {
  const { data, onItemClick, iconSize } = props;
  const { items } = data;

  return (
    <StyledMenuList className={props.className}>
      {items.map((item, index) =>
        (
          <li
            key={item.id}
          >
            <div
              role="button"
              tabIndex={index}
              onClick={onItemClick && ((e) => onItemClick(item, index, e))}
              onKeyUp={onItemClick && ((e) => onItemClick(item, index, e))}
            >
              <Flex flex="1 1 auto" alignItems="center">
                <Box p={2}>
                  {renderIcon(item.type, iconSize)}
                </Box>
                <Box p={2} ml="auto" width={1}>
                  <span>{item.name}</span>
                </Box>
              </Flex>
            </div>
          </li>
        ))}
    </StyledMenuList>
  );
};

AttachmentMenu.defaultProps = {
  onItemClick: null,
  className: 'LT-AttachmentMenu-Container',
};

AttachmentMenu.propTypes = {
  /**
   * Array of element to display as Action items [{ id: '1', name: 'MyAction'}]
   */
  data: PropTypes.object.isRequired,
  /**
   * Extra className to style the component
   */
  className: PropTypes.string,
  /**
   * Size for the icons
   */
  iconSize: PropTypes.number,
  /**
   * Callback function called when user clicks on an action Item
   * The callback has the following signature:
   * callback (action, index, e)
   */
  onItemClick: PropTypes.func,
};

AttachmentMenu.defaultProps = {
  className: 'LT-AttachmentMenu-Wrapper',
  iconSize: 20,
};

AttachmentMenu.displayName = 'AttachmentMenu';

export default AttachmentMenu;
