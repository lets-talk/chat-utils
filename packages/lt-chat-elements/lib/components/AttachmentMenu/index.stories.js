import React from 'react';
// Storybook stuff
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { withKnobs, number, select, object } from '@storybook/addon-knobs';

// Component to show on storybook
import AttachmentMenu from '.';
import { WrapWithTheme, defaultTheme, themeOptions } from '../../utils/stories';
import withTests from '../../test-utils/storiesWithTest';

const stories = storiesOf('AttachmentMenu', module);
stories.addDecorator(withKnobs);
stories.addDecorator(withTests('AttachmentMenu'));

const data = {
  items: [
    {
      id: 1,
      name: 'Imagen',
      type: 'image',
    },
    {
      id: 2,
      name: 'Video',
      type: 'video',
    },
    {
      id: 3,
      name: 'Audio',
      type: 'audio',
    },
    {
      id: 4,
      name: 'Documento',
      type: 'document',
    },
    {
      id: 5,
      name: 'Link de verificaciÃ³n',
      type: 'link',
    },
  ],
};

stories.addWithInfo(
  'AttachmentMenu',
  'This is the basic usage with the AttachmentMenu',
  () => (
    <div style={{ position: 'fixed', bottom: 0, right: '10px' }}>
      <WrapWithTheme
        themeName={select('Theme', themeOptions, defaultTheme)}
      >
        <AttachmentMenu
          onClickAction={action('onClick')}
          iconSize={number('IconSize', 20)}
          data={object('Data', data)}
        />
      </WrapWithTheme>
    </div>
  )
);
