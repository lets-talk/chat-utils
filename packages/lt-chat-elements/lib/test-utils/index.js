// test-utils/index.js

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import { ThemeProvider } from 'styled-components';

const defaultTheme = {
  palette: {
    common: {
      black: 'rgb(0, 0, 0)',
      white: 'rgb(255, 255, 255)',
      online: 'rgb(67, 160, 71)',
      offline: 'rgb(229, 57, 53)',
      sleeping: 'rgb(253, 216, 53)',
    },
    colors: {
      primary: {
        base: 'rgb(4, 38, 73)',
        light: 'rgb(86, 165, 246)',
        dark: 'rgb(0, 0, 0)',
      },
      accent: {
        base: 'rgb(255, 87, 50)',
        light: 'rgb(255, 171, 152)',
        dark: 'rgb(203, 37, 0)',
      },
      foreground: {
        base: 'rgb(94, 124, 139)',
        light: 'rgb(151, 173, 184)',
        dark: 'rgb(53, 70, 78)',
      },
      background: {
        base: 'rgb(255, 255, 255)',
        light: 'rgb(255, 255, 255)',
        dark: 'rgb(204, 204, 204)',
      },
    },
    contrastColors: {
      dark: {
        primary: 'rgb(255, 255, 255)',
        secondary: 'rgba(255, 255, 255, 0.7)',
        disabled: 'rgba(255, 255, 255, 0.5)',
        hint: 'rgba(255, 255, 255, 0.12)',
      },
      light: {
        primary: 'rgba(0, 0, 0, 0.87)',
        secondary: 'rgba(0, 0, 0, 0.54)',
        disabled: 'rgba(0, 0, 0, 0.38)',
        hint: 'rgba(0, 0, 0, 0.12)',
      },
    },
    error: {
      light: '#e57373',
      main: '#f44336',
      dark: '#d32f2f',
      contrastText: '#fff',
    },
    divider: 'rgba(255, 255, 255, 0.1)',
    action: {
      active: 'rgba(255, 255, 255, 0.9)',
      hover: 'rgba(255, 255, 255, 0.08)',
      selected: 'rgba(255, 255, 255, 0.14)',
      disabled: 'rgba(255, 255, 255, 0.26)',
      disabledBackground: 'rgba(255, 255, 255, 0.12)',
    },
  },
  shadows: {
    '2p': '0 2px 2px 0 rgba(0, 0, 0, 0.14), 0 3px 1px -2px rgba(0, 0, 0, 0.2), 0 1px 5px 0 rgba(0, 0, 0, 0.12)',
    '3p': '0 3px 4px 0 rgba(0, 0, 0, 0.14), 0 3px 3px -2px rgba(0, 0, 0, 0.2), 0 1px 8px 0 rgba(0, 0, 0, 0.12)',
    '4p': '0 4px 5px 0 rgba(0, 0, 0, 0.14), 0 1px 10px 0 rgba(0, 0, 0, 0.12), 0 2px 4px -1px rgba(0, 0, 0, 0.2)',
    '6p': '0 6px 10px 0 rgba(0, 0, 0, 0.14), 0 1px 18px 0 rgba(0, 0, 0, 0.12), 0 3px 5px -1px rgba(0, 0, 0, 0.2)',
    '8p': '0 8px 10px 1px rgba(0, 0, 0, 0.14), 0 3px 14px 2px rgba(0, 0, 0, 0.12), 0 5px 5px -3px rgba(0, 0, 0, 0.2)',
    '16p': '0 16px 24px 2px rgba(0, 0, 0, 0.14), 0 6px 30px 5px rgba(0, 0, 0, 0.12), 0 8px 10px -5px rgba(0, 0, 0, 0.2)',
  },
  typography: {
    fontFamily: "'Nimbus sans-serif', 'Sans', 'Helvetica', 'Arial'",
    fontSize: 'calc(1 * 1px)',
    weights: {
      fontWeightUltraThin: 100,
      fontWeightThin: 300,
      fontWeightNormal: 300,
      fontWeightMedium: 500,
      fontWeightSemibold: 600,
      fontWeightBold: 700,
    },
    classes: {
      body: {
        fontSize: 'calc(16 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      caption: {
        fontSize: 'calc(12 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      footnote: {
        fontSize: 'calc(14 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      small: {
        fontSize: 'calc(10 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      headline: {
        fontSize: 'calc(24 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      subhead: {
        fontSize: 'calc(18 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      title: {
        fontSize: 'calc(20 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      jumbo: {
        fontSize: 'calc(44 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      mega: {
        fontSize: 'calc(56 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
    },
  },
  components: {
    conversationBox: {
      externalBackgroundColor: 'rgba(94, 124, 139, 0.1)',
      internalBackgroundColor: 'rgba(255, 213, 0, 0.2)',
      importantBackgroundColor: 'rgb(255, 246, 244)',
    },
    loader: {
      backgroundColor: 'rgb(252, 88, 48)',
    },
  },
};

const shallowWithTheme = (component) => {
  const context = shallow(<ThemeProvider theme={defaultTheme} />)
    .instance()
    .getChildContext();
  return shallow(component, {
    context,
    childContextTypes: ThemeProvider.childContextTypes,
  });
};

const mountWithTheme = (component) => {
  const context = shallow(<ThemeProvider theme={defaultTheme} />)
    .instance()
    .getChildContext();

  return mount(component, {
    context,
    childContextTypes: ThemeProvider.childContextTypes,
  });
};

const renderWithTheme = (component) => renderer.create(<ThemeProvider theme={defaultTheme}>{component}</ThemeProvider>);

const staticWithTheme = (component) => render(<ThemeProvider theme={defaultTheme}>{component}</ThemeProvider>);

export {
  defaultTheme,
  shallowWithTheme,
  mountWithTheme,
  staticWithTheme,
  renderWithTheme,
};
