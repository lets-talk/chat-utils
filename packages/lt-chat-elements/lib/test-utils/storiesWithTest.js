import { withTests } from '@storybook/addon-jest';
import jestTestResults from '../../tests-results/jest-test-results.json';

export default withTests({ results: jestTestResults }, {
  filesExt: '__tests__/**.js',
});
