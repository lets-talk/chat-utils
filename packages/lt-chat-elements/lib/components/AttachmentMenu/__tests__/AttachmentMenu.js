import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';

import { renderWithTheme, staticWithTheme, shallowWithTheme } from '../../../test-utils';
import AttachmentMenu from '../index';

describe('AttachmentMenu component', () => {
  const emptyData = {
    items: [],
  };

  const data = {
    items: [
      {
        id: 1,
        name: 'Imagen',
        type: 'image',
      },
      {
        id: 2,
        name: 'Video',
        type: 'video',
      },
      {
        id: 3,
        name: 'Audio',
        type: 'audio',
      },
      {
        id: 4,
        name: 'Documento',
        type: 'document',
      },
      {
        id: 5,
        name: 'Link de verificaciÃ³n',
        type: 'link',
      },
    ],
  };

  it('should render with empty data', () => {
    const component = shallow(<AttachmentMenu data={emptyData} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with data', () => {
    const component = shallow(<AttachmentMenu data={data} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should have the correct styles (Those that depends on the theme)', () => {
    const tree = renderWithTheme(<AttachmentMenu data={data} />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('background-color', 'rgb(255,255,255)');
    expect(tree).toHaveStyleRule('font-size', 'calc(12 * 1px)');
    expect(tree).toHaveStyleRule('line-height', 'calc(16 * 1px)');
    expect(tree).toHaveStyleRule('font-weight', '500');
  });

  it('should render the correct html structure', () => {
    const wrapper = staticWithTheme(<AttachmentMenu data={data} />);
    expect(wrapper.find('li')).toHaveLength(5);
  });

  it('should handle the click event on the items correctly', () => {
    const onItemClick = jest.fn();
    const fakeEventObject = { preventDefault() {}, stopPropagation() {} };
    // const wrapper = mountWithTheme(<AttachmentMenu data={data} onItemClick={onItemClick} />);
    const wrapper = shallowWithTheme(<AttachmentMenu data={data} onItemClick={onItemClick} />);
    wrapper.find('li').last().find('div').first().simulate('click');
    // The function is called with the item clicked, the index and the event (item, index, e)
    expect(onItemClick).toHaveBeenCalledWith(data.items[4], 4, undefined);

    onItemClick.mockClear();
    wrapper.find('li').first().find('div').first().simulate('click');
    // The function is called with the item clicked, the index and the event (item, index, e)
    expect(onItemClick).toHaveBeenCalledWith(data.items[0], 0, undefined);

    onItemClick.mockClear();
    wrapper.find('li').at(1).find('div').first().simulate('click', fakeEventObject);
    // The function is called with the item clicked, the index and the event (item, index, e)
    expect(onItemClick).toHaveBeenCalledWith(data.items[1], 1, fakeEventObject);
  });

  it('should render ok with default icons', () => {
    const defaultIconData = {
      items: [
        {
          id: 1,
          name: 'Imagen',
          type: 'NOT_EXISTENT_TYPE',
        },
      ],
    };
    const component = shallow(<AttachmentMenu data={defaultIconData} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });
});
