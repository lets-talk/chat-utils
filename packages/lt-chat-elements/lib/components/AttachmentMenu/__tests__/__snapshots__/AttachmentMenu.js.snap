// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AttachmentMenu component should have the correct styles (Those that depends on the theme) 1`] = `
.c2 {
  box-sizing: border-box;
  padding: 8px;
}

.c3 {
  box-sizing: border-box;
  width: 100%;
  margin-left: auto;
  padding: 8px;
}

.c1 {
  box-sizing: border-box;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  display: block;
  cursor: pointer;
  padding: 0;
  margin: 0;
  border-radius: 5px;
  background-color: rgb(255,255,255);
  font-size: calc(12 * 1px);
  line-height: calc(16 * 1px);
  font-weight: 500;
  box-shadow: 0 4px 5px 0 rgba(0,0,0,0.14),0 1px 10px 0 rgba(0,0,0,0.12),0 2px 4px -1px rgba(0,0,0,0.2);
}

.c0 li {
  list-style: none;
  padding: 15px;
  color: rgba(0,0,0,0.54);
}

.c0 li:hover {
  background-color: rgba(0,0,0,0.38);
}

<ul
  className="LT-AttachmentMenu-Wrapper c0"
>
  <li
    onClick={undefined}
    onKeyUp={undefined}
  >
    <div
      className="c1"
    >
      <div
        className="c2"
      >
        <svg
          fill="currentColor"
          height={20}
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": undefined,
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 40 40"
          width={20}
        >
          <g>
            <path
              d="m14.1 22.5l-5.7 7.5h23.2l-7.5-10-5.7 7.5z m20.9 9.1q0 1.4-1 2.4t-2.4 1h-23.2q-1.4 0-2.4-1t-1-2.4v-23.2q0-1.4 1-2.4t2.4-1h23.2q1.4 0 2.4 1t1 2.4v23.2z"
            />
          </g>
        </svg>
      </div>
      <div
        className="c3"
      >
        <span>
          Imagen
        </span>
      </div>
    </div>
  </li>
  <li
    onClick={undefined}
    onKeyUp={undefined}
  >
    <div
      className="c1"
    >
      <div
        className="c2"
      >
        <svg
          fill="currentColor"
          height={20}
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": undefined,
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 40 40"
          width={20}
        >
          <g>
            <path
              d="m20 24.1l10-7.5-10-7.5v15z m13.4-20.7c1.8 0 3.2 1.4 3.2 3.2v20c0 1.8-1.4 3.4-3.2 3.4h-20c-1.8 0-3.4-1.6-3.4-3.4v-20c0-1.8 1.6-3.2 3.4-3.2h20z m-26.8 6.6v23.4h23.4v3.2h-23.4c-1.8 0-3.2-1.4-3.2-3.2v-23.4h3.2z"
            />
          </g>
        </svg>
      </div>
      <div
        className="c3"
      >
        <span>
          Video
        </span>
      </div>
    </div>
  </li>
  <li
    onClick={undefined}
    onKeyUp={undefined}
  >
    <div
      className="c1"
    >
      <div
        className="c2"
      >
        <svg
          fill="currentColor"
          height={20}
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": undefined,
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 40 40"
          width={20}
        >
          <g>
            <path
              d="m20 5h11.6v5h-6.6v18.4h-0.1c-0.4 3.7-3.6 6.6-7.4 6.6-4.1 0-7.5-3.4-7.5-7.5s3.4-7.5 7.5-7.5c0.9 0 1.7 0.2 2.5 0.5v-15.5z"
            />
          </g>
        </svg>
      </div>
      <div
        className="c3"
      >
        <span>
          Audio
        </span>
      </div>
    </div>
  </li>
  <li
    onClick={undefined}
    onKeyUp={undefined}
  >
    <div
      className="c1"
    >
      <div
        className="c2"
      >
        <svg
          fill="currentColor"
          height={20}
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": undefined,
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 40 40"
          width={20}
        >
          <g>
            <path
              d="m14.1 22.5l-5.7 7.5h23.2l-7.5-10-5.7 7.5z m20.9 9.1q0 1.4-1 2.4t-2.4 1h-23.2q-1.4 0-2.4-1t-1-2.4v-23.2q0-1.4 1-2.4t2.4-1h23.2q1.4 0 2.4 1t1 2.4v23.2z"
            />
          </g>
        </svg>
      </div>
      <div
        className="c3"
      >
        <span>
          Documento
        </span>
      </div>
    </div>
  </li>
  <li
    onClick={undefined}
    onKeyUp={undefined}
  >
    <div
      className="c1"
    >
      <div
        className="c2"
      >
        <svg
          fill="currentColor"
          height={20}
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": undefined,
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 40 40"
          width={20}
        >
          <g>
            <path
              d="m30 33.4v-16.8h-20v16.8h20z m0-20c1.8 0 3.4 1.4 3.4 3.2v16.8c0 1.8-1.6 3.2-3.4 3.2h-20c-1.8 0-3.4-1.4-3.4-3.2v-16.8c0-1.8 1.6-3.2 3.4-3.2h15.2v-3.4c0-2.8-2.4-5.2-5.2-5.2s-5.2 2.4-5.2 5.2h-3.2c0-4.6 3.8-8.4 8.4-8.4s8.4 3.8 8.4 8.4v3.4h1.6z m-10 15c-1.8 0-3.4-1.6-3.4-3.4s1.6-3.4 3.4-3.4 3.4 1.6 3.4 3.4-1.6 3.4-3.4 3.4z"
            />
          </g>
        </svg>
      </div>
      <div
        className="c3"
      >
        <span>
          Link de verificaci√≥n
        </span>
      </div>
    </div>
  </li>
</ul>
`;

exports[`AttachmentMenu component should render with data 1`] = `
<styled.ul
  className="LT-AttachmentMenu-Wrapper"
>
  <li
    key="1"
  >
    <Flex
      alignItems="center"
      flex="1 1 auto"
      theme={
        Object {
          "breakpoints": Array [
            "40em",
            "52em",
            "64em",
          ],
          "fontSizes": Array [
            12,
            14,
            16,
            20,
            24,
            32,
            48,
            64,
            96,
            128,
          ],
          "space": Array [
            0,
            4,
            8,
            16,
            32,
            64,
            128,
            256,
            512,
          ],
        }
      }
    >
      <Box
        p={2}
        theme={
          Object {
            "breakpoints": Array [
              "40em",
              "52em",
              "64em",
            ],
            "fontSizes": Array [
              12,
              14,
              16,
              20,
              24,
              32,
              48,
              64,
              96,
              128,
            ],
            "space": Array [
              0,
              4,
              8,
              16,
              32,
              64,
              128,
              256,
              512,
            ],
          }
        }
      >
        <MdImage
          size={20}
        />
      </Box>
      <Box
        ml="auto"
        p={2}
        theme={
          Object {
            "breakpoints": Array [
              "40em",
              "52em",
              "64em",
            ],
            "fontSizes": Array [
              12,
              14,
              16,
              20,
              24,
              32,
              48,
              64,
              96,
              128,
            ],
            "space": Array [
              0,
              4,
              8,
              16,
              32,
              64,
              128,
              256,
              512,
            ],
          }
        }
        width={1}
      >
        <span>
          Imagen
        </span>
      </Box>
    </Flex>
  </li>
  <li
    key="2"
  >
    <Flex
      alignItems="center"
      flex="1 1 auto"
      theme={
        Object {
          "breakpoints": Array [
            "40em",
            "52em",
            "64em",
          ],
          "fontSizes": Array [
            12,
            14,
            16,
            20,
            24,
            32,
            48,
            64,
            96,
            128,
          ],
          "space": Array [
            0,
            4,
            8,
            16,
            32,
            64,
            128,
            256,
            512,
          ],
        }
      }
    >
      <Box
        p={2}
        theme={
          Object {
            "breakpoints": Array [
              "40em",
              "52em",
              "64em",
            ],
            "fontSizes": Array [
              12,
              14,
              16,
              20,
              24,
              32,
              48,
              64,
              96,
              128,
            ],
            "space": Array [
              0,
              4,
              8,
              16,
              32,
              64,
              128,
              256,
              512,
            ],
          }
        }
      >
        <MdVideoLibrary
          size={20}
        />
      </Box>
      <Box
        ml="auto"
        p={2}
        theme={
          Object {
            "breakpoints": Array [
              "40em",
              "52em",
              "64em",
            ],
            "fontSizes": Array [
              12,
              14,
              16,
              20,
              24,
              32,
              48,
              64,
              96,
              128,
            ],
            "space": Array [
              0,
              4,
              8,
              16,
              32,
              64,
              128,
              256,
              512,
            ],
          }
        }
        width={1}
      >
        <span>
          Video
        </span>
      </Box>
    </Flex>
  </li>
  <li
    key="3"
  >
    <Flex
      alignItems="center"
      flex="1 1 auto"
      theme={
        Object {
          "breakpoints": Array [
            "40em",
            "52em",
            "64em",
          ],
          "fontSizes": Array [
            12,
            14,
            16,
            20,
            24,
            32,
            48,
            64,
            96,
            128,
          ],
          "space": Array [
            0,
            4,
            8,
            16,
            32,
            64,
            128,
            256,
            512,
          ],
        }
      }
    >
      <Box
        p={2}
        theme={
          Object {
            "breakpoints": Array [
              "40em",
              "52em",
              "64em",
            ],
            "fontSizes": Array [
              12,
              14,
              16,
              20,
              24,
              32,
              48,
              64,
              96,
              128,
            ],
            "space": Array [
              0,
              4,
              8,
              16,
              32,
              64,
              128,
              256,
              512,
            ],
          }
        }
      >
        <MdAudiotrack
          size={20}
        />
      </Box>
      <Box
        ml="auto"
        p={2}
        theme={
          Object {
            "breakpoints": Array [
              "40em",
              "52em",
              "64em",
            ],
            "fontSizes": Array [
              12,
              14,
              16,
              20,
              24,
              32,
              48,
              64,
              96,
              128,
            ],
            "space": Array [
              0,
              4,
              8,
              16,
              32,
              64,
              128,
              256,
              512,
            ],
          }
        }
        width={1}
      >
        <span>
          Audio
        </span>
      </Box>
    </Flex>
  </li>
  <li
    key="4"
  >
    <Flex
      alignItems="center"
      flex="1 1 auto"
      theme={
        Object {
          "breakpoints": Array [
            "40em",
            "52em",
            "64em",
          ],
          "fontSizes": Array [
            12,
            14,
            16,
            20,
            24,
            32,
            48,
            64,
            96,
            128,
          ],
          "space": Array [
            0,
            4,
            8,
            16,
            32,
            64,
            128,
            256,
            512,
          ],
        }
      }
    >
      <Box
        p={2}
        theme={
          Object {
            "breakpoints": Array [
              "40em",
              "52em",
              "64em",
            ],
            "fontSizes": Array [
              12,
              14,
              16,
              20,
              24,
              32,
              48,
              64,
              96,
              128,
            ],
            "space": Array [
              0,
              4,
              8,
              16,
              32,
              64,
              128,
              256,
              512,
            ],
          }
        }
      >
        <MdImage
          size={20}
        />
      </Box>
      <Box
        ml="auto"
        p={2}
        theme={
          Object {
            "breakpoints": Array [
              "40em",
              "52em",
              "64em",
            ],
            "fontSizes": Array [
              12,
              14,
              16,
              20,
              24,
              32,
              48,
              64,
              96,
              128,
            ],
            "space": Array [
              0,
              4,
              8,
              16,
              32,
              64,
              128,
              256,
              512,
            ],
          }
        }
        width={1}
      >
        <span>
          Documento
        </span>
      </Box>
    </Flex>
  </li>
  <li
    key="5"
  >
    <Flex
      alignItems="center"
      flex="1 1 auto"
      theme={
        Object {
          "breakpoints": Array [
            "40em",
            "52em",
            "64em",
          ],
          "fontSizes": Array [
            12,
            14,
            16,
            20,
            24,
            32,
            48,
            64,
            96,
            128,
          ],
          "space": Array [
            0,
            4,
            8,
            16,
            32,
            64,
            128,
            256,
            512,
          ],
        }
      }
    >
      <Box
        p={2}
        theme={
          Object {
            "breakpoints": Array [
              "40em",
              "52em",
              "64em",
            ],
            "fontSizes": Array [
              12,
              14,
              16,
              20,
              24,
              32,
              48,
              64,
              96,
              128,
            ],
            "space": Array [
              0,
              4,
              8,
              16,
              32,
              64,
              128,
              256,
              512,
            ],
          }
        }
      >
        <MdLockOpen
          size={20}
        />
      </Box>
      <Box
        ml="auto"
        p={2}
        theme={
          Object {
            "breakpoints": Array [
              "40em",
              "52em",
              "64em",
            ],
            "fontSizes": Array [
              12,
              14,
              16,
              20,
              24,
              32,
              48,
              64,
              96,
              128,
            ],
            "space": Array [
              0,
              4,
              8,
              16,
              32,
              64,
              128,
              256,
              512,
            ],
          }
        }
        width={1}
      >
        <span>
          Link de verificaci√≥n
        </span>
      </Box>
    </Flex>
  </li>
</styled.ul>
`;

exports[`AttachmentMenu component should render with empty data 1`] = `
<styled.ul
  className="LT-AttachmentMenu-Wrapper"
/>
`;
